src/software/jetson_nano/services/robot_auto_test.cpp

```diff
@@ -0,0 +1,109 @@
+#include <chrono>
+#include <thread>
+
+#include "proto/message_translation/tbots_geometry.h"
+#include "proto/primitive/primitive_msg_factory.h"
+#include "shared/2021_robot_constants.h"
+#include "shared/constants.h"
+#include "software/jetson_nano/primitive_executor.h"
+#include "software/jetson_nano/services/motor.h"
+#include "software/jetson_nano/services/power.h"
+#include "software/util/scoped_timespec_timer/scoped_timespec_timer.h"
+
+extern "C"
+{
+#include "external/trinamic/tmc/ic/TMC4671/TMC4671.h"
+#include "external/trinamic/tmc/ic/TMC4671/TMC4671_Variants.h"
+#include "external/trinamic/tmc/ic/TMC6100/TMC6100.h"
+}
+
+std::unique_ptr<MotorService> motor_service_;
+std::unique_ptr<PowerService> power_service_;
+RobotConstants_t robot_constants_;
+int read_value;
+
+// SPI Chip Selects
+static const uint8_t FRONT_LEFT_MOTOR_CHIP_SELECT =
+    motor_service_->FRONT_LEFT_MOTOR_CHIP_SELECT;
+static const uint8_t FRONT_RIGHT_MOTOR_CHIP_SELECT =
+    motor_service_->FRONT_RIGHT_MOTOR_CHIP_SELECT;
+static const uint8_t BACK_LEFT_MOTOR_CHIP_SELECT =
+    motor_service_->BACK_LEFT_MOTOR_CHIP_SELECT;
+static const uint8_t BACK_RIGHT_MOTOR_CHIP_SELECT =
+    motor_service_->BACK_RIGHT_MOTOR_CHIP_SELECT;
+
+static const uint8_t CHIP_SELECT[] = {
+    FRONT_LEFT_MOTOR_CHIP_SELECT, FRONT_RIGHT_MOTOR_CHIP_SELECT,
+    BACK_LEFT_MOTOR_CHIP_SELECT, BACK_RIGHT_MOTOR_CHIP_SELECT};
+
+constexpr int ASCII_4671_IN_HEXADECIMAL = 0x34363731;
+constexpr double THRESHOLD              = 0.0001;
+constexpr int DELAY                     = 10000;
+std::string runtime_dir                 = "/tmp/tbots/yellow_test";
+
+int main(int argc, char **argv)
+{
+    LoggerSingleton::initializeLogger(runtime_dir);
```

Comment:
you should use the NetworkLoggerSingleton here. Refer to the `thunderloop.cpp`

Reply:
Updated, however not too sure if we need to a redis client for this test so I defaulted the values

Reply:
okay my bad let's revert for now. I didn't realize at first that you were streaming logs back to the host computer using ansible

Reply:
done